java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -480,27 +482,53 @@ private void executeMethodInvocation(MethodInvocationTree mit) {
     setSymbolicValueOnFields(mit);
     // unstack arguments and method identifier
     ProgramState.Pop unstack = programState.unstackValue(mit.arguments().size() + 1);
-    programState = unstack.state;
     logState(mit);
 
+    programState = unstack.state;
+    // Enqueue exceptional paths
+    node.programPoint.block.exceptions().forEach(b -> enqueue(new ExplodedGraph.ProgramPoint(b, 0), programState, !b.isCatchBlock()));
+
     // get method behavior for method with known declaration (ie: within the same file)
-    Tree declaration = mit.symbol().declaration();
+    MethodBehavior methodInvokedBehavior = null;
+    Symbol methodSymbol = mit.symbol();
+    Tree declaration = methodSymbol.declaration();
     if(declaration != null) {
-      // (FIXME) nothing done of the invoked behavior
-      MethodBehavior methodInvokedBehavior = symbolicExecutionVisitor.execute((MethodTree) declaration);
+      methodInvokedBehavior = symbolicExecutionVisitor.execute((MethodTree) declaration);
+    }
+    if (methodInvokedBehavior != null && !methodSymbol.isAbstract()) {
+      List<SymbolicValue> invocationArguments = invocationArguments(unstack.values);
+      methodInvokedBehavior.yields()
+        .stream()
+        .flatMap(yield -> yield.statesAfterInvocation(invocationArguments, programState, () -> constraintManager.createMethodSymbolicValue(mit, unstack.values)).stream())
+        .forEach(ps -> {
+          ProgramState programState = ps;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "programState" which hides the field declared at line 107. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)